pilz_trajectory_generation/test/unittest_trajectory_blender_transition_window.cpp

```diff
@@ -525,6 +568,200 @@ TEST_P(TrajectoryBlenderTransitionWindowTest, testLINLINBlending)
 
 }
 
+/**
+ * @brief  Tests the blending of two cartesian linear trajectories with a non-trivial overlap using robot model
+ * The test sequence is repeated twice, using robot model with and without gripper
+ * Test Sequence:
+ *    1. Generate two linear trajectories from the test data set, such that the second one is the reversed first one.
+ *    2. Generate blending trajectory.
+ *    3. Check blending trajectory:
+ *      - for position, velocity, and acceleration bounds,
+ *      - for continuity in joint space,
+ *      - for continuity in cartesian space.
+ *
+ * Expected Results:
+ *    1. Two linear trajectories generated.
+ *    2. Blending trajectory generated.
+ *    3. No bound is violated, the trajectories are continuous
+ *        in joint and cartesian space.
+ */
+TEST_P(TrajectoryBlenderTransitionWindowTest, testLINLINBlendingOverlappingTrajectories)
+{
+  auto test_data = test_data_.front();
+
+  // modify end position such that trajectories overlap
+  test_data.end_position = test_data.start_position;
+
+  // generate two lin trajectories from test dataset
+  planning_interface::MotionPlanResponse res_lin_1, res_lin_2;
+  double dis_lin_1, dis_lin_2;
+  ASSERT_TRUE(testutils::generateTrajFromBlendTestData(robot_model_,
+                                                       lin_,
+                                                       planning_group_,
+                                                       target_link_,
+                                                       test_data,
+                                                       sampling_time_, sampling_time_,
+                                                       res_lin_1, res_lin_2,
+                                                       dis_lin_1, dis_lin_2))
+      << "Failed to generate LIN trajectories from test data";
+
+  // blend two lin trajectories and check the result
+  pilz::TrajectoryBlendRequest blend_req;
+  pilz::TrajectoryBlendResponse blend_res;
+
+  blend_req.group_name = planning_group_;
+  blend_req.link_name = target_link_;
+  blend_req.first_trajectory = res_lin_1.trajectory_;
+  blend_req.second_trajectory = res_lin_2.trajectory_;
+  blend_req.blend_radius = 0.5*dis_lin_1;
+  EXPECT_TRUE(blender_->blend(blend_req, blend_res));
+
+  // check the blend result
+  EXPECT_TRUE(testutils::checkBlendResult(blend_req,
+                                          blend_res,
+                                          planner_limits_,
+                                          joint_velocity_tolerance_,
+                                          joint_acceleration_tolerance_,
+                                          cartesian_velocity_tolerance_,
+                                          cartesian_angular_velocity_tolerance_));
+}
+
+/**
+ * @brief  Tests the blending of two cartesian trajectories generated by adding sine curves to linear trajectories
```

Comment:
Could you explain a bit further the reasoning behind this?

Reply:
The whole idea is to blend trajectories, that differ from a straight line. I changed this line. The procedure of adding the sine is explained further down in the comment anyways.